src/language-js/index.js

```diff
@@ -6,12 +6,14 @@ const estreeJsonPrinter = require("./printer-estree-json");
 const options = require("./options");
 
 const languages = [
+  // @ts-ignore - TBD (...)
```

Comment:
I have one question, isn't it just json object? What exactly need to do?

As I said I don't use JSDoc, maybe this is a silly question.

Reply:
I would get the following error:

```
src/language-js/index.js:9:26 - error TS2307: Cannot find module 'linguist-languages/data/JavaScript' or its corresponding type declarations.

9   createLanguage(require("linguist-languages/data/JavaScript"), (data) => ({
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I think it would be a pure JSON-like object, yes.

I am now wondering if there is something we could do to improve the way the language-specific data would be exported by `linguist-languages` and imported by this module.

Reply:
This link https://github.com/microsoft/TypeScript/issues/7071  says  json  should not  be  an error, maybe  we  can  add `.json` extension?

Reply:
I tried using the JSON extension in all cases except one, hope it will be green.

I am still wondering if we could improve the import of the data from linguist-languages somehow:)

Reply:
@brodybits It works? Can you send another PR to fix all `linguist-languages` import? It will be weird if we only import it with `.json` in js-related file, @evilebottnawi and me can have a quick review and merge it.

Reply:
Also it would be great to enable eslint rule to force `.json` extension.

Reply:
> I am still wondering if we could improve the import of the data from linguist-languages somehow:)

I think import needed JSON file is already the best way?

Reply:
I will contribute the eslint config & fix `linguist-languages` later today. Nice find.

Reply:
Great, in that case, I'm going to sleep now. 0:32 in my place :smile:

Reply:
That hint to use eslint-plugin-import did not work for me. I suspect that is because we are using require to import the JSON items from `linguist-languages`. There is actually not such a large amount of work to start using typecheck on the other language sources, so I will work on this instead.